modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp

```diff
@@ -1464,18 +1481,576 @@ void run_sepfilter3x3_impl(DST out[], const SRC *in[], int width, int chan,  \
     }                                                                        \
 }
 
-RUN_SEPFILTER3X3_IMPL(uchar , uchar )
-RUN_SEPFILTER3X3_IMPL( short, uchar )
-RUN_SEPFILTER3X3_IMPL( float, uchar )
+RUN_SEPFILTER3X3_IMPL(uchar, uchar)
+RUN_SEPFILTER3X3_IMPL(short, uchar)
+RUN_SEPFILTER3X3_IMPL(float, uchar)
 RUN_SEPFILTER3X3_IMPL(ushort, ushort)
-RUN_SEPFILTER3X3_IMPL( short, ushort)
-RUN_SEPFILTER3X3_IMPL( float, ushort)
-RUN_SEPFILTER3X3_IMPL( short,  short)
-RUN_SEPFILTER3X3_IMPL( float,  short)
-RUN_SEPFILTER3X3_IMPL( float,  float)
+RUN_SEPFILTER3X3_IMPL(short, ushort)
+RUN_SEPFILTER3X3_IMPL(float, ushort)
+RUN_SEPFILTER3X3_IMPL(short, short)
+RUN_SEPFILTER3X3_IMPL(float, short)
+RUN_SEPFILTER3X3_IMPL(float, float)
 
 #undef RUN_SEPFILTER3X3_IMPL
 
+//-----------------------------
+//
+// Fluid kernels: sepFilter 5x5
+//
+//-----------------------------
+
+#if CV_SIMD
+
+// this code with manually vectored rounding to uchar
+template<bool noscale, typename SRC>
+static void run_sepfilter5x5_any2char(uchar out[], const SRC *in[], int width, int chan,
+                                      const float kx[], const float ky[], int border,
+                                      float scale, float delta,
+                                      float *buf[], int y, int y0)
+{
+    constexpr int kxLen = 5;
+    constexpr int kyLen = kxLen;
+    constexpr int buffSize = 5;
+
+    int r[buffSize];
+    for (int n = 0; n < buffSize; ++n)
+    {
+        r[n] = (y - y0 + n) % 5;  // previous, this, next rows
+    }
+
+    const int length = width * chan;
+    const int shift = chan;
+
+    // horizontal pass
+
+    int k0 = (y == y0) ? 0 : 3;
```

Comment:
Probably:
```c++
int k0 = (y == y0) ? 0 : 3;
```
The idea of `k0` is to decide if you need to process all **5** rows in the buffer or just the last one.  
Here you process only the last one if the previous rows were already done at previous passes.